src/deadline/job_attachments/download.py

```diff
@@ -86,20 +86,45 @@
 def get_manifest_from_s3(
     manifest_key: str, s3_bucket: str, session: Optional[boto3.Session] = None
 ) -> BaseAssetManifest:
+    _, manifest = get_asset_root_and_manifest_from_s3(manifest_key, s3_bucket, session)
+    return manifest
+
+
+def get_asset_root_and_manifest_from_s3(
+    manifest_key: str, s3_bucket: str, session: Optional[boto3.Session] = None
+) -> Tuple[Optional[str], BaseAssetManifest]:
     s3_client = get_s3_client(session=session)
     try:
-        file_buffer = io.BytesIO()
-        s3_client.download_fileobj(
-            s3_bucket,
-            manifest_key,
-            file_buffer,
-            ExtraArgs={"ExpectedBucketOwner": get_account_id(session=session)},
+        # Attempt to get the manifest and metadata in a single request. If it's larger than a single 8MB
+        # chunk, fetch it in chunks using the S3 transfer manager.
```

Comment:
It looks like 8MB gets downloaded twice in the larger case. I wonder if it would be better to not fall back to the S3 transfer manager, and instead only wrap parallelism around multi-manifest downloads?

Reply:
I did some scans of manifests that I've generated, and the largest one I found was 1.3MB with about 9000 files. It was generated by the job https://github.com/aws-deadline/deadline-cloud-samples/tree/mainline/job_bundles/copy_s3_prefix_to_job_attachments. The largest output in tests I ran was 600KB with 4000 files.

A larger case might be a manifest of 1 million files, with data per path of 256 bytes. That would be about 256MB. Based on that, I think 5GB as an upper limit on manifest size is reasonable, and simplifying the code to not have the fallback would be good.